nixos/tests/hydra/default.nix

```diff
@@ -0,0 +1,75 @@
+import ../make-test.nix ({ pkgs, ...} :
+
+let
+   trivialJob = pkgs.writeTextDir "trivial.nix" ''
+     with import <nix/config.nix>;
+
+     { trivial = builtins.derivation {
+         name = "trivial";
+         system = "x86_64-linux";
+         PATH = coreutils;
+         builder = shell;
+         args = ["-c" "touch $out"];
+       };
+     }
+   '';
+
+    createTrivialProject = pkgs.stdenv.mkDerivation {
+      name = "create-trivial-project";
+      unpackPhase = ":";
+      buildInputs = [ pkgs.makeWrapper ];
+      installPhase = "install -m755 -D ${./create-trivial-project.sh} $out/bin/create-trivial-project.sh";
+      postFixup = ''
+        wrapProgram "$out/bin/create-trivial-project.sh" --prefix PATH ":" ${pkgs.stdenv.lib.makeBinPath [ pkgs.curl ]} --set EXPR_PATH ${trivialJob}
+      '';
+    };
+
+in {
+  name = "hydra-init-localdb";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ pstn lewo ];
+  };
+
+  machine =
+    { config, pkgs, ... }:
+
+    {
+      virtualisation.memorySize = 1024;
+      environment.systemPackages = [ createTrivialProject pkgs.jq ];
+      services.hydra = {
+        enable = true;
+
+        #Hydra needs those settings to start up, so we add something not harmfull.
+        hydraURL = "example.com";
+        notificationSender = "example@example.com";
+      };
+      nix = {
+        buildMachines = [{
+          hostName = "localhost";
+          systems = [ "x86_64-linux" ];
+        }];
+      };
+    };
+
+  testScript =
+    ''
+      # let the system boot up
+      $machine->waitForUnit("multi-user.target");
+      # test whether the database is running
+      $machine->succeed("systemctl status postgresql.service");
+      # test whether the actual hydra daemons are running
+      $machine->succeed("systemctl status hydra-queue-runner.service");
+      $machine->succeed("systemctl status hydra-init.service");
+      $machine->succeed("systemctl status hydra-evaluator.service");
+      $machine->succeed("systemctl status hydra-send-stats.service");
+
+      $machine->succeed("hydra-create-user admin --role admin --password admin");
+
+      # create a project with a trivial job
+      $machine->waitForOpenPort(3000);
+      $machine->succeed("create-trivial-project.sh");
+
+      # make sure the build as been successfully built
+      $machine->waitUntilSucceeds('curl -L -s http://localhost:3000/build/1 -H "Accept: application/json" |  jq .buildstatus | xargs test 0 -eq');
```

Comment:
The test seems non-deterministic on my local machine: it succeeds the first time but then fails on repeated runs, because the jobset is unchanged and skipped by hydra, so it waits here until the `waitUntilSucceeds` times out .
You could fix this by artifically changing the derivation hash of the `trivial` derivation on every test run, for instance by making some attribute  depend on current time or a random value.

Reply:
It is also sometimes failing on my machine... When it fails, in the test log, there is:
```
machine# [   47.021598] hydra-queue-runner[1167]: marking build 1 as failed
```
And in the log of the build is:
```
<3>builder for '/nix/store/nfmhzq2sv51qn1mh0lfmkpiwpk8dc2v5-trivial.drv' failed due to signal 9 (Killed)
```
But the output path as been produced even if the job is marked as failed. It seems this is related to Hydra and not to Nix.
Note also if I restart the job, it succeeds...

Reply:
@xeji I've fixed the non deterministic issue I encountered. Could you try again?

Reply:
Thanks, non-deterministic failure seems gone here too. Looks good now.